src/alert/alert.tsx

```diff
@@ -126,7 +128,7 @@ export const Alert = ({
             $hasActionLink={!!actionLink}
         >
             <div ref={contentRef}>{children}</div>
-            {actionLink && renderLink()}
+            {renderLink()}
```

Comment:
Typescript can't recognise that we have already performed the type check

in most cases I resolve this by:
1. passing the checked prop to be an argument
2. moving the check into the render function